src/lib/select/select.html

```diff
@@ -15,11 +15,26 @@
   <div class="mat-select-arrow-wrapper"><div class="mat-select-arrow"></div></div>
 </div>
 
+<ng-template #panelTemplate>
+  <div
+    #panel
+    class="mat-select-panel {{ _getPanelTheme() }}"
+    [ngClass]="panelClass"
+    [@transformPanel]="_getPanelAnimationState()"
+    (@transformPanel.done)="_panelDoneAnimatingStream.next($event.toState)"
+    [style.transformOrigin]="_transformOrigin"
+    [style.font-size.px]="_triggerFontSize"
+    (keydown)="_handleKeydown($event)">
+    <ng-content></ng-content>
```

Comment:
Can we avoid doing the "ng-content inside ng-template` trick? It ends up always instantiating the content, even when it's not going to be attached.

Reply:
We could make it work by allowing people to provide an `ng-template` as the content. The problem is that we need the content to be rendered, in order to know what the options inside the select are. We need to know the options for a lot of things like showing the selected value, knowing where to position the dropdown, supporting keyboard navigation while closed etc.